### PR TITLE
Function to get token balance of a ERC20 contract

### PR DESCRIPTION
**Pull request by gabrielskm**

### CODE REVIEW

File: EthereumKit/Helper/ERC20+Contract.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+                return "0x70a08231"
+            }
+        }
+        
+        func data() -> String {
+            switch self {
+            case .balanceOf(let address):
+                let noHexAddress = address.stripHexPrefix()
+                let padding = String(Array(repeating: "0", count: 64 - noHexAddress.count)) // Padding the 20 bytes token address to 32 bytes with 0
+                return self.methodSignature + padding + noHexAddress
```
Comment (yuzushioh) [inline EthereumKit/Helper/ERC20+Contract.swift]: `func pad(string: String) -> String` is available in ERC20 class. Can you use it here please?

File: EthereumKit/Helper/ERC20+Contract.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
+extension ERC20 {
+    enum ContractFunctions {
+        case balanceOf(address: String)
+
```
Comment (yuzushioh) [inline EthereumKit/Helper/ERC20+Contract.swift]: Can you add Transfer case here please? its defined in ERC20 by `transferSigniture` property

File: EthereumKit/Helper/ERC20+Contract.swift (hunk 1 above)
Comment (yuzushioh) [reply EthereumKit/Helper/ERC20+Contract.swift]: `func generateDataParameter(toAddress: String, amount: String) throws -> Data` is also available for generating data parameter for a contract call

File: EthereumKit/Helper/ERC20+Contract.swift (hunk 2 above)
Comment (gabrielskm) [reply EthereumKit/Helper/ERC20+Contract.swift]: Refactored this enum to handle also the transfer method as well

**Comment by gabrielskm** [comment]
@yuzushioh Changes done in commit 6477606561a92d7c88d213d10c7451309f0817bc please review

File: EthereumKit/Networking/Geth+Contract.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                    completionHandler(.failure(EthereumKitError.convertError(.failedToConvert(hexBalance))))
+                    return
+                }
+                let balance = Balance(wei: wei)
+                completionHandler(.success(balance))
+                break
+            case .failure(let error):
+                completionHandler(.failure(error))
+                break
+            }
```
Comment (yuzushioh) [inline EthereumKit/Networking/Geth+Contract.swift]: You dont necessarily put `break` here if theres no logic to be executed after `switch` section

File: EthereumKit/Networking/Geth+Contract.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+        self.call(to: contract.contractAddress, data: data.toHexString().addHexPrefix()) { result in
+            
+            switch result {
+            case .success(let hexBalance):
+                guard let wei = Wei(hexBalance.lowercased().stripHexPrefix(), radix: 16) else {
+                    completionHandler(.failure(EthereumKitError.convertError(.failedToConvert(hexBalance))))
+                    return
+                }
+                let balance = Balance(wei: wei)
```
Comment (yuzushioh) [inline EthereumKit/Networking/Geth+Contract.swift]: I prefer writing like this. I think it gives more readability.

```swift
if let  let wei = Wei(hexBalance.lowercased().stripHexPrefix(), radix: 16) {
    let balance = Balance(wei: wei)
    completionHandler(.success(balance))
} else {
    completionHandler(.failure(EthereumKitError.convertError(.failedToConvert(hexBalance))))
}
```

File: EthereumKit/Networking/Geth+Contract.swift (hunk 3 above)
Comment (gabrielskm) [reply EthereumKit/Networking/Geth+Contract.swift]: break statements removed

File: EthereumKit/Networking/Geth+Contract.swift (hunk 4 above)
Comment (gabrielskm) [reply EthereumKit/Networking/Geth+Contract.swift]: you are right, it is clearer, i changed it

**Comment by gabrielskm** [comment]
Thanks for the feedback again, I fixed the getTokenBalance code to follow your comments in ddf1f2d23511c0bb220aeac2ec5071ab3f009753

**Pull request merged**